server/graphql/loaders/expenses.ts

```diff
@@ -144,3 +145,25 @@ export const generateExpensesSecurityCheckLoader = req => {
     },
   );
 };
+
+export const expenseAgreementsLoader = new DataLoader<number, Agreement[]>(async (expenseIds: number[]) => {
```

Comment:
Similarly to https://github.com/opencollective/opencollective-api/pull/8894, we should use a generator function here